lldb/packages/Python/lldbsuite/test/dpu/dpu_list/host.c

```diff
@@ -1,26 +1,19 @@
 #include <dpu.h>
 
 int main() {
-  struct dpu_rank_t *dpu_rank;
+  struct dpu_set_t set;
+  struct dpu_set_t dpu;
 
-  if (dpu_alloc("backend=hw,cycleAccurate=true", &dpu_rank) != DPU_API_SUCCESS) {
-    if (dpu_alloc("backend=simulator", &dpu_rank) != DPU_API_SUCCESS)
-    	return -1;
+  if (dpu_alloc(1, "backend=hw,cycleAccurate=true", &set) != DPU_API_SUCCESS) {
+    DPU_ASSERT(dpu_alloc(1, "backend=simulator", &set));
   }
 
-  struct dpu_t *dpu = dpu_get(dpu_rank, 0, 0);
-
-  if (dpu_load_individual(dpu, DPU_EXE) != DPU_API_SUCCESS) {
-    dpu_free(dpu_rank);
-    return -2;
+  DPU_FOREACH(set, dpu) {
```

Comment:
Why using DPU_FOREACH when the set is composed of 1 element?

Reply:
It was not useful here indeed.